implementations/rust/ockam/ockam_api/src/session/util.rs

```diff
@@ -5,19 +5,44 @@ use std::time::Duration;
 pub(crate) const MAX_RECOVERY_TIME: Duration = Duration::from_secs(10);
 pub(crate) const MAX_CONNECT_TIME: Duration = Duration::from_secs(5);
 
-pub(crate) fn starts_with_host_tcp(addr: &MultiAddr) -> Option<usize> {
+pub(crate) fn starts_with_host_tcp(addr: &MultiAddr) -> Option<(MultiAddr, MultiAddr)> {
     let host_match = Match::any([DnsAddr::CODE, Ip4::CODE, Ip6::CODE]);
     if addr.matches(0, &[host_match, Tcp::CODE.into()]) {
-        Some(2)
+        Some(addr.split(1))
     } else {
         None
     }
 }
 
-pub(crate) fn starts_with_secure(addr: &MultiAddr) -> Option<usize> {
+pub(crate) fn starts_with_secure(addr: &MultiAddr) -> Option<(MultiAddr, MultiAddr)> {
     if addr.matches(0, &[Secure::CODE.into()]) {
-        Some(1)
+        Some(addr.split(1))
     } else {
         None
     }
 }
+#[cfg(test)]
+mod tests {
+    use ockam_multiaddr::MultiAddr;
+    use crate::session::util::starts_with_host_tcp;
+
+    #[test]
+    fn starts_with_host_tcp_returns_distinct_pair() {
+
+
+        let m = MultiAddr::try_from("ip4/127.0.0.1/tcp/1234").unwrap();
+        let (m1, m2) = starts_with_host_tcp(&m).unwrap();
+        assert!(m1.to_string() != m2.to_string());
```

Comment:
We can improve this test a little with checking for the correct starting address, and then the remaining address.

Keeping from the example above - `/dnsaddr/localhost/tcp/4000/service/api`

m1 -> `/dnsaddr/localhost/tcp/4000`

m2 -> `/service/api`